### PR TITLE
Revisiting Log creation and report

### PR DESCRIPTION
**Pull request by Mxrcon**
Hey :wave:, I revisited our logging and worked for a generation of more clearer logs using PODE custom logs. These logs could be used for fast debugging on Heroku terminal (That's why i changed from file to terminal log).

Here's an example of a log generated by These changes:
```terminal
[REQUEST] HOST: 127.0.0.1 - STATUS: 200 - METHOD: GET - RESOURCE: /v1/search - PROTOCOL: HTTP/1.1 - AGENT: restish-0.14.0
[REQUEST] HOST: 127.0.0.1 - STATUS: 200 - METHOD: GET - RESOURCE: /v1/search - PROTOCOL: HTTP/1.1 - AGENT: Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0
[REQUEST] HOST: 127.0.0.1 - STATUS: 200 - METHOD: GET - RESOURCE: / - PROTOCOL: HTTP/1.1 - AGENT: Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0
[REQUEST] HOST: 127.0.0.1 - STATUS: 200 - METHOD: GET - RESOURCE: /docs/view - PROTOCOL: HTTP/1.1 - AGENT: Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0
[REQUEST] HOST: 127.0.0.1 - STATUS: 200 - METHOD: GET - RESOURCE: /docs - PROTOCOL: HTTP/1.1 - AGENT: Mozilla/5.0 (X11; Linux x86_64; rv:103.0) Gecko/20100101 Firefox/103.0
[REQUEST] HOST: 127.0.0.1 - STATUS: 404 - METHOD: GET - RESOURCE: /v1/search - PROTOCOL: HTTP/1.1 - AGENT: restish-0.14.0

```

With this structure we can find 404 status using grep or other filtering command line tools.

Also, we should revisit this logging again once I have an answer here https://github.com/Badgerati/Pode/issues/1012

I kindly ask for your review on this.

### CODE REVIEW

**Comment by abhi18av** [comment]
Task linked: [CU-2zard5d Finalize the logging logic](https://app.clickup.com/t/2zard5d)

**Comment by Mxrcon** [comment]
Hmm, I decided to take another design path: Let's keep request log separated from the search results, as for the success of a request doesn't matter if we find or not the requested package. I'll create another custom log and modularize the logs and separate them from sucuri. Using this, I'll concentrate the log on the pode layer on the api and rely on it for throwing logs and errors.

**Comment by abhi18av** [comment]
>  Let's keep request log separated from the search results

I like this idea - worth tracking this as a separate task 👍 


For the current PR, I think the minimal viable task is to simply capture and output the requested package/channels in the same log file.

File: Sucuri-API/Public/Start-SucuriServer.ps1 [hunk 1]
```diff
@@ -44,14 +44,27 @@ function Start-SucuriServer {
...
 
-      $logPath = (Join-Path (Split-Path -Path $PSScriptRoot -Parent) -ChildPath "logs")
+      # Request Logging
+      $Request_LogMethod = New-PodeLoggingMethod -Custom -ScriptBlock {
 
-      #  Enabling logging
+      param($item)
 
-      New-PodeLoggingMethod -File -Path "$logPath\requestlogs" -Name 'requests' -MaxDays 1 | Enable-PodeRequestLogging
+      "[REQUEST] HOST: $($item.Host) - STATUS: $($item.Response.StatusCode) - METHOD: $($item.Request.Method) - RESOURCE: $($item.Request.Resource) - PROTOCOL: $($item.Request.Protocol) - AGENT: $($item.Request.Agent)" | Out-Default
```
[the remark this answers is not in the document]
Comment (Mxrcon) [reply Sucuri-API/Public/Start-SucuriServer.ps1:52]: hm you mean printing the json to CLI? do you think that's will be readable?

[the remark this answers is not in the document]
Comment (abhi18av) [reply Sucuri-API/Public/Start-SucuriServer.ps1:52]: Not to the CLI, but to the `stdout` (normally the terminal) yes.

> do you think that's will be readable?

That's a good question, generally speaking we don't read the entire log, we search it for specific events. Searching across plain text log is tougher than searching across JSON data.

[the remark this answers is not in the document]
Comment (Mxrcon) [reply Sucuri-API/Public/Start-SucuriServer.ps1:52]: I can easily transform this to a json as the `$log_Item` is curently a hashtable with the values and could be converted to json with `Convert-ToJson` command. But I'm worried about the user experience reading the logs as it'll end in this output at the terminal:

```terminal
Start-SucuriServer
Listening on the following 1 endpoint(s) [1 thread(s)]:
	- http://localhost:8085/
{"LogType": "Search","LogInformation": [{"package": "prokka","channels": "bioconda","status": "OK"}]}

{"LogType": "Request","LogInformation": [{"host": "127.0.0.1","status": "200","method": "GET","resource": "/v1/search","protocol": "HTTP/1.1","agent": "restish-0.14.0"}]}



```

**Comment by Mxrcon** [comment]
Erorr log example (Conda binary not found) : 
```terminal
Start-SucuriServer             
Listening on the following 1 endpoint(s) [1 thread(s)]:
	- http://localhost:8085/
[ERROR] SERVER: linux-A515-54 - CATEGORY: OperationStopped: (There's no conda-co…e binary available.:String) [], RuntimeException - MESSAGE: There's no conda-compatible binary available. - STACK-TRACE: at FindCondaBinary, /home/davi_marcon/github/biosharp/sucuri-api/Sucuri-API/Private/Utils/FindCondaBinary.ps1: line 20
at Get-Version, /home/davi_marcon/github/biosharp/sucuri-api/Sucuri-API/Public/Get-Version.ps1: line 33
at <ScriptBlock>, /home/davi_marcon/github/biosharp/sucuri-api/Sucuri-API/Private/Api-Versions/v1Search.ps1: line 15
at Invoke-PodeScriptBlock, /home/davi_marcon/.local/share/powershell/Modules/Pode/2.7.1/Public/Utilities.ps1: line 634
at <ScriptBlock>, <No file>: line 106
[SEARCH] PACKAGE: prokka - CHANNELS: bioconda - STATUS: Not Found

[REQUEST] HOST: 127.0.0.1 - STATUS: 404 - METHOD: GET - RESOURCE: /v1/search - PROTOCOL: HTTP/1.1 - AGENT: restish-0.14.0
Terminating... Done

```

Common log example: 

```terminal 
[SEARCH] PACKAGE: prokka - CHANNELS: bioconda - STATUS: OK

[REQUEST] HOST: 127.0.0.1 - STATUS: 200 - METHOD: GET - RESOURCE: /v1/search - PROTOCOL: HTTP/1.1 - AGENT: restish-0.14.0

```

File: Sucuri-API/Public/Start-SucuriServer.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (abhi18av) [reply Sucuri-API/Public/Start-SucuriServer.ps1]: I think this is looking great!

If its not a big effort, then let's use an environment variables called `SUCURI_API_LOGGING_FORMAT`, with default value as `json` this way we can control the app via the env.

File: Sucuri-API/Private/Api-Versions/v1Search.ps1 [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+     }
+      $Search_Log | Write-PodeLog -Name 'Searches'
+   }
+
+   else {
+      Write-PodeJsonResponse -Value @{'packageName' = $pkg_name; 'latestVersion' = $null; 'channels' = $null } -StatusCode 404
+      $Search_Log = @{
+         Package = $WebEvent.Query['pkg_name']
+         Channels  = $WebEvent.Query['channels']
+         Status = "Not Found"
```
Comment (abhi18av) [inline Sucuri-API/Private/Api-Versions/v1Search.ps1:35]:
```suggestion
         Status = "404"

```

File: Sucuri-API/Private/Api-Versions/v1Search.ps1 [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+   catch { $_ | Write-PodeErrorLog }
+
+   if ( $response -match '\d\.\d\.\d|\d\.\d' ) {
+
+      Write-PodeJsonResponse -Value @{'packageName' = $pkg_name; 'latestVersion' = $response ; 'channels'= $channels } -StatusCode 200
+      
+      $Search_Log = @{
+         Package = $WebEvent.Query['pkg_name']
+         Channels  = $WebEvent.Query['channels']
+         Status = "OK"
```
Comment (abhi18av) [inline Sucuri-API/Private/Api-Versions/v1Search.ps1:25]:
```suggestion
         Status = "200"

```

**Pull request merged**